packages/shared/src/components/MobileHeaderRankProgress.tsx

```diff
@@ -1,25 +1,45 @@
-import React, { CSSProperties, ReactElement } from 'react';
+import React, {
+  CSSProperties,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  useContext,
+} from 'react';
 import useReadingRank from '../hooks/useReadingRank';
 import RankProgressWrapper from './RankProgressWrapper';
 import { rankToColor } from '../lib/rank';
+import ProgressiveEnhancementContext from '../contexts/ProgressiveEnhancementContext';
+
+interface WrapperProps extends HTMLAttributes<HTMLDivElement> {
+  children?: ReactNode;
+}
+const Wrapper = ({ children, ...props }: WrapperProps) => {
+  return (
+    <div className="w-8 h-8" {...props}>
+      {children}
+    </div>
+  );
+};
 
 export default function MobileHeaderRankProgress(): ReactElement {
+  const { windowLoaded } = useContext(ProgressiveEnhancementContext);
```

Comment:
Why do we wait for `windowLoaded` here?
Once the rank is available we can show it

Reply:
This causes a react leak same as for the sidebar.
Not sure if it can be solved otherwise but would love to hear alternatives.

Reply:
What do you mean by "react leak"?